pkg/api/handlers/utils/images.go

```diff
@@ -1,19 +1,50 @@
 package utils
 
 import (
+	"encoding/base64"
+	"encoding/json"
 	"fmt"
 	"net/http"
+	"strings"
 
 	"github.com/containers/image/v5/docker"
 	"github.com/containers/image/v5/storage"
 	"github.com/containers/image/v5/transports/alltransports"
 	"github.com/containers/image/v5/types"
 	"github.com/containers/libpod/libpod"
 	"github.com/containers/libpod/libpod/image"
+	dockerAPITypes "github.com/docker/docker/api/types"
 	"github.com/gorilla/schema"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 )
 
+// XRegistryAuthHeader is the key to the encoded registry authentication
+// configuration in http requests for the Docker compatibility endpoints.
+const XRegistryAuthHeader = "X-Registry-Auth"
+
+// GetXRegistryAuth extracts a DockerAuthConfig from the request's header if
+// present. Encoding errors are logged but yield nil (for Docker compat).
+func GetXRegistryAuth(r *http.Request) *types.DockerAuthConfig {
+	authHeader := r.Header.Get(XRegistryAuthHeader)
+	authConfig := &dockerAPITypes.AuthConfig{}
+	if len(authHeader) > 0 {
+		authJSON := base64.NewDecoder(base64.URLEncoding, strings.NewReader(authHeader))
+		if err := json.NewDecoder(authJSON).Decode(authConfig); err != nil {
+			logrus.Errorf("Error parsing %q header: %v", XRegistryAuthHeader, err)
```

Comment:
I think it would be better to return a 400 to the client than ignore the header.

Reply:
SGTM
Just to be extra sure: Docker is also just logging which is the motivation behind that. Propagating the error is much cleaner though.